README.md

```diff
@@ -6,76 +6,215 @@
+
+Contribution workflow:
+1. Fork the project.
+2. Get the [development environment running](TODO:hyperlink).
+3. Make your feature addition or bug fix.
+4. Add [tests for it](TODO:hyperlink). This is important so we don't break it in a future version unintentionally.
+5. Send a pull request. The tests will run on it automatically, so don't worry if you couldn't get them running locally.
+
+Please refer to the [CONTRIBUTING][contributing] guidelines for more information.
+
+### New Ecosystems
+
+Currently, the Dependabot team is not accepting support for new ecosystems. We are prioritising upgrades to already supported ecosystems at this time.
+
+Please refer to the [CONTRIBUTING][contributing] guidelines for more information.
+
+# Development Guide
 
-Dependabot Core is a collection of packages for automating dependency updating
-in Ruby, JavaScript, Python, PHP, Elixir, Elm, Go, Rust, Java and
-.NET. It can also update git submodules, Docker files, and Terraform files.
-Highlights include:
+## Getting a Development Environment Going
 
-- Logic to check for the latest version of a dependency *that's resolvable given
-  a project's other dependencies*
-- Logic to generate updated manifest and lockfiles for a new dependency version
-- Logic to find changelogs, release notes, and commits for a dependency update
+The first step to debugging a problem or writing a new feature is getting a development environment going. We provide a
+custom [Docker-based](https://www.docker.com/products/docker-desktop/) developer shell that bakes in all required dependencies.
+In most cases this is the best way to work with the project.
 
-## Other Dependabot resources
+The developer shell uses volume mounts to incorporate your local changes to Dependabot's source code. This way you can
+edit locally using your favorite editor and the changes are immediately reflected within the docker container for performing
+[dry-runs](TODO:hyperlink) or executing [tests](TODO:Hyperlink).
+Note: See caveat about [editing the native package manager helper scripts](TODO:Hyperlink).
 
-In addition to this library, you may be interested in the [dependabot-script][dependabot-script] repo,
-which provides a collection of scripts that use this library to update dependencies on GitHub Enterprise, GitLab,
-BitBucket or Azure DevOps.
+### Quickstart
 
-## Setup
+The script to launch the developer shell builds the docker images from scratch if it can't find them locally. This can take a while.
 
-To run all of Dependabot Core, you'll need Ruby, Python, PHP, Elixir, Node, Go,
-Elm, and Rust installed. However, if you just wish to run it for a single
-language you can get away with just having that language and Ruby.
+Skip the wait by pulling the pre-built image for the ecosystem you want to work on. The image name uses the [YAML ecosystem name](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem)
```

Comment:
We used to have this shortcut buried further down, but we want new contributors to get going as fast as possible, so highlighting how to pull the image as the first step will save them 5-10 mins and lead to a much more pleasant first experience. At least for me it did.